src/Cart/Cart.php

```diff
@@ -37,7 +38,15 @@ class Cart
     public function __construct(Session $session = null)
     {
         $this->session = $session;
-        $this->items = collect($this->session ? $this->session->get('shop.cart', []) : []);
+        $this->items = collect([]);
+
+        $sessionItems = $this->session ? $this->session->get('shop.cart', []) : [];
+        $this->session->remove('shop.cart');
+
+        foreach ($sessionItems as $sessionItem) {
+            $refreshedPackage = Package::where("id", "=", $sessionItem->id)->get()->first();
+            $this->add($refreshedPackage, $sessionItem->quantity);
+        }
```

Comment:
This code is not really optimized as it will make a database query for each item in the cart 😕 Also a cart can contain any `Buyable` implementation like `Offer` but this change will break this behavior.

Also using `get()->first()` will fetch all the rows in the table and then take the first one. It can be easily replaced by just `first()` to only fetch the first record of the table (and there is also `firstWhere(...)` as an shortcut for `where(...)->first()` in Laravel 7)